apps/worker/tasks/base.py

```diff
@@ -232,6 +234,67 @@ def _retry(self, countdown=None):
             if TestResultsFlow.has_begun():
                 TestResultsFlow.log(TestResultsFlow.UNCAUGHT_RETRY_EXCEPTION)
 
+    def _get_total_attempts(self) -> int:
+        """
+        Get total attempts including re-deliveries from visibility timeout expiration.
+
+        When visibility timeout expires, Redis re-queues the task but Celery treats
+        it as a new delivery (not a retry), so request.retries doesn't increment.
+        We track total attempts in task headers to catch these re-deliveries.
+
+        Returns:
+            Total number of attempts (retries + re-deliveries)
+        """
+        if not hasattr(self, "request"):
+            return 0
+
+        # Get retry count from Celery
+        retry_count = self.request.retries if hasattr(self.request, "retries") else 0
+
+        # Get total attempts from headers (tracks re-deliveries)
+        headers = self.request.get("headers", {}) or {}
+        total_attempts_header = headers.get("total_attempts", None)
```

Comment:
if we aren't using `headers` anywhere else here, why is this broken into 2 lines?